.github/workflows/Readme.md

```diff
@@ -0,0 +1,297 @@
+- **PAT Token**: PAT token is used for checking the runner label and their availability. 
+## Variables and Secrets Configuration
+
+The workflow requires certain variables and secrets to be set in your repository settings:
+
+### Variables (`vars`)
+To configure the required variables, navigate to your repository's Settings > Secrets and variables > Actions > Variables, then add each variable (e.g., DOCKERHUB_USERNAME, QUAYIO_USERNAME) with their corresponding values.
+
+- `DOCKERHUB_USERNAME`: Your Docker Hub username.
+- `QUAYIO_USERNAME`: Your Quay.io username.
+- `ACACIA_BUCKETNAME`: Your targeted acacia bucketname.
+
+### Secrets (`secrets`)
+To configure the required variables, navigate to your repository's Settings > Secrets and variables > Actions > Variables, then add each variable (e.g., ACACIA_ACCESS_KEY_ID, ACACIA_SECRET_ACCESS_KEY, DOCKERHUB_TOKEN, QUAYIO_TOKEN) with their corresponding values.
+- `DOCKERHUB_TOKEN`: A Docker Hub access token with permissions to push images.
+- `QUAYIO_TOKEN`: A Quay.io access token with permissions to push images.
+- `ACACIA_ACCESS_KEY_ID`: Access key ID for the S3-compatible storage.
+- `ACACIA_SECRET_ACCESS_KEY`: Secret access key for the S3-compatible storage.
+- `PAT_TOKEN`: PAT token is used for checking the runner label and their availability. 
+
+These variables and secrets are used throughout the workflow for authentication and tagging purposes.
+
+### Environments (`manual_approval`)
+The manual_approval environment is configured to require explicit manual approval before the deployment process proceeds. This ensures that all deployments are reviewed and authorized by designated team members, adding an additional layer of security and control to the CI pipeline.
+
+Navigate to your repository's Settings > Environments > New Environment > Name (`manual_approval`), then under Deployment protection rules, click (`Require reviewers`) and Save protection rules.
+## Workflow Overview
+
+The workflow consists of several jobs that execute sequentially or in parallel, depending on their dependencies.
+
+### PREPARE-job
+
+This initial job sets up the environment and determines whether the workflow should proceed based on the trigger logic.
+
+**Key Steps:**
+
+1. **Checkout Repository**: Uses `actions/checkout@v4` to fetch the latest code with a `fetch-depth` of 2 to ensure sufficient commit history for comparison.
+
+2. **Get Changed Files**: 
+   - Uses `git diff` to identify Dockerfiles (`*.dockerfile`) that have changed between the previous commit and the current one.
+   - Stores the list of changed files in the `files` output variable.
+
+3. **Set Proceed Flag**: 
+   - Checks if no Dockerfiles or multiple Dockerfiles have been modified.
+     - If no Dockerfiles are modified, sets `proceed_valid` to `false` and exits.
+     - If multiple Dockerfiles are modified, sets `proceed_valid` to `false` and exits.
```

Comment:
This is where some logic updates might be necessary later as we shouldn't treat this the same as no Dockerfiles modified. I'll need to think about how best to handle this exit.

Reply:
OK, this version will keep it. After our discussion, we can change its behaviours.